File: Semana 2/FibonacciLastDigit.cpp
#include <iostream>
#include <vector>
using namespace std;


long long fastFibonacci(long long n){
    if(n < 2){
        return n;
    }
    long long aux1 = 0;
    long long aux2 = 1;
    long long numero = 1;
    for (long long i  = 0; i < n-1 ; i ++){
        aux2 = numero;
        numero = (numero + aux1)%10;
        aux1 = aux2;
    }
    return numero;
}
int main(){
    long long number;
    cin >> number;
    cout << fastFibonacci(number)<<endl;
}

File: Semana 2/FibonacciLastDigit.py
def fastFibonacci(number):
    if number < 2:
        return number
    aux = 0
    aux1 = 0
    result = 1
    for i in range(1,number):
        aux = result
        result = result+aux1
        aux1 = aux
    return result

if __name__ == "__main__":
    number = int(input())
    print(fastFibonacci(number)%10)